src/js/components/search/filters/contractFilters/ContractFilter.jsx

```diff
@@ -0,0 +1,135 @@
+/**
+ * ContractFilter.jsx
+ * Created by Emily Gullo on 6/23/2017
+ */
+
+import React from 'react';
+import * as Icons from 'components/sharedComponents/icons/Icons';
+import * as ContractFieldDefinitions from 'dataMapping/search/contractFields';
+
+import PrimaryCheckboxType from 'components/sharedComponents/checkbox/PrimaryCheckboxType';
+
+const propTypes = {
+    toggleFilter: React.PropTypes.func,
+    contractFilterType: React.PropTypes.string,
+    contractFilterOptions: React.PropTypes.string,
+    contractFilterState: React.PropTypes.string
+};
+
+const defaultShown = 5;
+
+const defaultState = {
+    shown: defaultShown,
+    shownType: 'more'
+};
+
+export default class ContractFilter extends React.Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = defaultState;
+
+        // Bind functions
+        this.toggleValue = this.toggleValue.bind(this);
+        this.generateContractFilters = this.generateContractFilters.bind(this);
+        this.toggleShownAmount = this.toggleShownAmount.bind(this);
+    }
+
+    toggleShownAmount() {
+        const contractFilters = ContractFieldDefinitions[this.props.contractFilterOptions];
+
+        let updatedState = defaultState;
+
+        if (this.state.shownType === 'more') {
+            updatedState = {
+                shown: Object.keys(contractFilters).length,
+                shownType: 'fewer'
+            };
+        }
+
+        this.setState(updatedState);
+    }
+
+    toggleValue(value) {
```

Comment:
Do we need this function? Can we just pass the prop straight down to `toggleCheckboxType`?